// electron/index.js
const {stub} = require('sinon');

exports.shell = {
    openExternal: stub().returns(true),
    openItem: stub().returns(true)
};

const browserWindow = {
    isFocused: stub().returns(true),
    show: stub(),
    focus: stub(),
};

const webContents = {
    openDevTools: stub()
};

const app = {
    getPath: stub().returns(''),
    dock: {
        setBadge: stub().returns(true),
    },
};

const defaultConfig = {
    notification: true,
    notification_sound: true,
    hotkey_accelerator: null,
    expand_tweet: 'always',
    plugin: [],
    mute: {
        home: true,
        mention: false,
    },
    max_timeline_items: null,
    proxy: null,
    sticky_window: false,
    caffeinated: false,
};

const getGlobal = stub();
getGlobal.withArgs('config').returns(defaultConfig);
getGlobal.withArgs('config_path').returns('');

exports.remote = {
    getCurrentWindow: stub().returns(browserWindow),
    getCurrentWebContents: stub().returns(webContents),
    app,
    getGlobal,
};

exports.ipcRenderer = {
    on: stub().returns(exports.ipcRenderer),
    removeListener: stub(),
};
